base/unicode/utf8proc.jl

```diff
@@ -156,8 +156,7 @@ iscntrl(c::Char) = (c <= Char(0x1f) || Char(0x7f) <= c <= Char(0x9f))
 ispunct(c::Char) = (UTF8PROC_CATEGORY_PC <= category_code(c) <= UTF8PROC_CATEGORY_PO)
 
 # \u85 is the Unicode Next Line (NEL) character
-# the check for \ufffd allows for branch removal on ASCIIStrings
-@inline isspace(c::Char) = c == ' ' || '\t' <= c <='\r' || c == '\u85' || '\ua0' <= c && c != '\ufffd' && category_code(c)==UTF8PROC_CATEGORY_ZS
+@inline isspace(c::Char) = c == ' ' || '\t' <= c <='\r' || c == '\u85' || '\ua0' <= c && category_code(c) == UTF8PROC_CATEGORY_ZS
```

Comment:
Note this method applies to any `Char`, not just those coming from `String`. So I'd keep the fast path for ASCII, which could still be useful for custom string types. Anyway, its cost should be negligible compared to the call to utf8proc.

A better fix (to apply after this PR) would be to use `<= 0x80` so that we don't call utf8proc for ASCII characters in general (i.e. at runtime).